lib/ash/type/type.ex

```diff
@@ -535,13 +535,20 @@ defmodule Ash.Type do
   end
 
   defp do_generator({:array, type}, constraints) do
-    generator = do_generator(type, item_constraints(constraints))
+    item_constraints = item_constraints(constraints)
+    generator = do_generator(type, item_constraints)
 
     generator =
       if constraints[:nil_items?] do
         StreamData.one_of([StreamData.constant(nil), generator])
       else
         generator
+        |> StreamData.filter(fn value ->
+          case Ash.Type.apply_constraints(type, value, item_constraints) do
+            {:ok, value} -> value != nil
+            _ -> false
```

Comment:
Last nitpick is that I think we should leave these in the stream, to produce some other unrelated error later down the road.